app.js

```diff
@@ -25,4 +25,49 @@ module.exports = {
   start: port => {
     app.listen(port, () => console.log(`server up: ${port}`));
   }
-};
\ No newline at end of file
+};
+
+app.post('/signup', (req, res) => {
+  users.save(req.body)
+    .then(user => {
+      let token = users.generateToken(user);
+      res.status(200).send(token);
+    })
+    .catch( err => res.status(403).send('error creating user'));
+});
+
+app.post('/signin', basicAuth, (req, res) => {
+  // req.token only exists because of our basic auth middleware
+  res.status(200).send(req.token);
+});
+
+// this is our "redirect_uri"
+app.get('/oauth', oauth, (req, res) => {
+  res.status(200).send(req.token);
```

Comment:
To DRY this code up a bit, you could turn that `res.status` into a function that you could call